### DISCUSSION TITLE
How to getSmiles() with version 2.2?

### DISCUSSION THREAD

**Discussion by ixdi** [epam/ketcher#614] [Q&A] [answered]
Hi, as documentation says, I can get the ketcher reference in js (via iframe) but an error occurs when trying to call
```
ketcher.getSmiles()
```
getSmiles is not defined. How can I get the smiles from the new version of ketcher 2.2?

Documentation -> https://lifescience.opensource.epam.com/ketcher/developers-manual.html

Note: I've tried both standalone and remote versions

Thanks

**Comment by AndreiMazol** [comment, marked as answer]
Dear @ixdi ,

This moment Ketcher is under redesign so we are focusing on code changes rather than documentation. Documentation will be updated a little bit later.

To get SMILES you need to use the following method:

`getSmilesAsync(isExtended: boolean = false): Promise<string>`

Example of usage:

`ketcher.getSmilesAsync().then(smiFile => console.log(smiFile))`

Best regards,
Andrei
